ament_package/templates.py

```diff
@@ -23,9 +23,18 @@
 IS_WINDOWS = os.name == 'nt'
 
 
-def get_environment_hook_template_path(name):
-    with importlib_resources.path('ament_package.template.environment_hook', name) as path:
+if hasattr(importlib_resources, 'files'):
+    def _get_path(template, name):
+        path = importlib_resources.files(template).joinpath(name)
         return str(path)
+else:
+    def _get_path(template, name):
+        with importlib_resources.path(template, name) as path:
+            return str(path)
```

Comment:
I'm going to suggest that you invert this so that the check is done inside of `_get_path` at runtime.  The main reason for this is to keep logic out of import time.  So something like:

```suggestion
def _get_path(template, name):
    if hasattr(importlib_resources, 'files'):
        return importlib_resources.files(template).joinpath(name)
    else:
        with importlib_resources.path(template, name) as path:
            return str(path)
```